packages/ui/src/SplitPage.tsx

```diff
@@ -1,80 +1,121 @@
-import React from "react";
-import {ListRenderItemInfo, View} from "react-native";
+import React, {Children, ReactChild, ReactElement, useState} from "react";
+import {ListRenderItemInfo, ScrollView, StyleSheet, View} from "react-native";
 
 import {Box} from "./Box";
-import {Color} from "./Common";
+import {Color, SPACING} from "./Common";
 import {FlatList} from "./FlatList";
 import {IconButton} from "./IconButton";
 import {mediaQueryLargerThan} from "./MediaQuery";
+import {SegmentedControl} from "./SegmentedControl";
 import {Spinner} from "./Spinner";
 import {Unifier} from "./Unifier";
 
 interface SplitPageProps {
+  children?: ReactChild | ReactChild[] | null;
+  tabs?: string[];
   // TODO: figure out navigation
-  navigation: any;
+  navigation?: any;
   loading?: boolean;
   color?: Color;
   keyboardOffset?: number;
-  rightButton?: string;
-  rightButtonOnClick?: () => void;
-  renderListViewItem: (itemInfo: ListRenderItemInfo<any>) => React.ReactElement | null;
-  renderListViewHeader?: () => React.ReactElement | null;
-  renderContent: (index?: number) => React.ReactElement | null;
+  renderListViewItem: (itemInfo: ListRenderItemInfo<any>) => ReactElement | null;
+  renderListViewHeader?: () => ReactElement | null;
+  renderContent?: (index?: number) => ReactElement | ReactElement[] | null;
   listViewData: any[];
   listViewExtraData?: any;
   listViewWidth?: number;
+  renderChild?: () => ReactChild;
 }
 
-interface SplitPageState {
-  selectedId?: number;
-}
+const defaultStyles = StyleSheet.create({
+  mainContentContainer: {
```

Comment:
You should inline these to match the rest of the component. I could see an argument for elementContainer being a stylesheet since it's longer and used in multiple places (could also skip the .create part and just have the style object, then use `style={elementContainer}`)